### DISCUSSION TITLE
Can I initialize the store with props during rendering, i.e. using a context-provider pattern?

### DISCUSSION THREAD

**Discussion by michaelschufi** [udecode/zustand-x#61] [General]
I'm looking for a similar approach as this https://docs.pmnd.rs/zustand/guides/initialize-state-with-props.

I need to initialize the store based on data from the server. In Next.js you can, for example, let the layout get some data from the server and use the Context-Provider pattern to initialize a context. (https://nextjs.org/docs/getting-started/react-essentials#using-context-in-client-components)

**Comment by financialvice** [comment]
@michaelschufi - curious if you've figured out a solution for this? 

I feel like I'm getting close by following this documentation from zustand, however I believe a different implementation of "useStore" would be required.

`export const useCounterStore = <T,>(
  selector: (store: CounterStore) => T,
): T => {
  const counterStoreContext = useContext(CounterStoreContext)

  if (!counterStoreContext) {
    throw new Error(`useCounterStore must be use within CounterStoreProvider`)
  }

  return useStore(counterStoreContext, selector)
}`
